src/Hintman/Suggest.hs

```diff
@@ -0,0 +1,44 @@
+{-# LANGUAGE NamedFieldPuns #-}
+
+module Hintman.Suggest
+       ( getTrailLines
+       , toLines
+       , suggest
+       ) where
+
+import Data.Text (stripEnd)
+
+
+data LineState = Plain | Trailing
+    deriving (Eq, Show)
+
+data Line = Line
+    { lineRaw   :: Int
+    , lineBody  :: Text
+    , lineState :: LineState
+    } deriving (Eq, Show)
+
+data Suggestion = Suggestion
+    { suggestionFile :: FilePath
+    , suggestionNew :: Text
+    , suggestionComment :: Maybe Text  -- ^ Optimal comment to the suggestion
+    }
+
+toLines :: Text -> [Line]
+toLines txt = map (\(n,x) -> Line {lineRaw = n, lineBody = x, lineState = lineState' x}) lTxt
+  where
+    lTxt :: [(Int, Text)]
+    lTxt = zip [1..] (lines txt)
+
+    lineState' :: Text -> LineState
+    lineState' x
```

Comment:
I feel like the `toLines` function is very powerful and other hints might make use of it. Could it make sense to leave the trailing space logic out of it? same for the `suggest` function below, which seems to be for trailing space only

Reply:
> to leave the trailing space logic out of it?

Sorry. Do you mean to make `toLines :: Text -> [Line]` with _Trailing_ lines only or ...?

Reply:
I was thinking that `toLines` could be a transformation function only (from Text to Line) and then have a separate function (suggestNoTrailingSpace?) to turn the Line into a Suggestion. This way other hints can use `toLines`. But it's just an idea :)

Reply:
> (from Text to Line)

Text has multiply lines, `Line` is about one row. 
But yes, I may code function like `suggest :: FilePath -> Text -> [Suggestion]`
Just a sec..